src/store/codeVault/getters.ts

```diff
@@ -6,7 +6,7 @@ const codeVaultGetters: GetterTree<CodeVaultState, RootState> = {
   files: (state) => state.files,
   filenames: (state) => {
     const names: Set<string> = new Set();
-    state.files.forEach((file) => names.add(file.filename));
+    state.files.forEach((file) => names.add(file.filename.slice(0, -3))); // remove '.py'
```

Comment:
It makes more sense to me to not remove the '.py' here and rather add the file extension in the place where we check uniqueness

Reply:
Lol I knew it would be that

Yeah so I guess I'll just add an `extension?: string` to the `uniqueTextInput` function?

Reply:
That seems like a good idea to me

Reply:
Or you could add another getter for filenames without extensions